File: vipaq/src/Binacle.ViPaq/Models/Width.cs
namespace Binacle.ViPaq;

// How many bytes one integer takes on the wire (PROTOCOL.md §4).
//
// Codes 2 and 3 are named but carry no width. They exist so the two spare codes in every width field are
// visible in the code, and so a decoder rejects them by name rather than by falling off a switch. An encoder
// must never write one.
internal enum Width
{
	Eight = 0,
	Sixteen = 1,
	Reserved2 = 2,
	Reserved3 = 3
}

File: vipaq/packages/binacle-vipaq/src/models/width.ts
// Ports C#: Width. How wide a section's values are on the wire (PROTOCOL.md §4). Only Eight and Sixteen ever
// reach the wire; codes 2 and 3 are reserved, so a decoder that reads one rejects the blob. The old BitSize had
// ThirtyTwo and SixtyFour too — they are gone: every interoperable value fits in 16 bits (0..65535).
export const enum Width {
	Eight = 0,
	Sixteen = 1,
	Reserved2 = 2,
	Reserved3 = 3,
}
